src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -213,4 +213,14 @@ public String getStubName(Interface service) {
   public String getCreateStubFunctionName(Interface service) {
     return getGrpcClientTypeName(service).replace(".", ".New");
   }
+
+  @Override
+  public String getStaticLangStreamingReturnTypeName(Method method, MethodConfig methodConfig) {
+    // Unsafe string manipulation: The name looks like "LibraryService_StreamShelvesClient",
+    // neither camel or underscore.
+    return String.format(
+        "%s_%sClient",
+        converter.getTypeName(method.getParent()).getNickname(),
+        method.getSimpleName());
```

Comment:
Done.

In Go, streaming is done with two stream objects, one sending and one receiving (if bidirectional, then both can send *and* receive). One of these objects is owned by the server and another is owned by the client. So the word "client" here doesn't mean "client library" but "the client half of the streaming object pair".